File: 121. Best Time to Buy and Sell Stock/solution.go
func maxProfit(prices []int) int {
    var profit int
    min := 65535

    for _,price := range(prices) {
        if price < min {
            min = price
            continue
        }

        if profit < price - min {
            profit = price - min
        }
    }

    return profit
}

File: 121. Best Time to Buy and Sell Stock/solution.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        if len(prices) < 2:
            return 0
        
        # run through all possible cases
        profit = 0
        price = prices[0]
        for i in range(1, len(prices)):
            profit = max(profit, prices[i] - price)
            price = min(price, prices[i])
            
        return profit
